### PR TITLE
fix(linux): html tags in package description in keyboard's About page

### LINKED ISSUE #4769: bug(linux): html tags in package description in keyboard's About page
**Describe the bug**

image

The description is shown with these html tags `<p>...</p>` when triggered it from the <kbd>About</kbd> button on the Keyman Configuration.

**Expected behavior**

The tags should not be shown explicitly.

---

**Keyman for Linux:**
 - OS: Ubuntu 20.04.2 LTS -->
 - Keyman version: 14.0.270
 - Target Application: Keyman

---

**Keyboard**
 - Keyboard name: khmer_angkor
 - Keyboard version: 1.0.7
 - Language name: Khmer

### PR DESCRIPTION
**Pull request by TheSchlunz**
This change fixes a bug where html tags were shown in the package description in a keyboard's About page.

Fixes #4769

# User Testing
## Preparations
  - install "khmer_angkor" keyboard 
 
## Tests

**TEST_NO_HTML_TAGS**: no html tags are shown in the package description in keyboard's About page
  - start `km-config`
  - select "Khmer Angkor" keyboard
  - click "About" button
  - verify that no html tags are shown in the package description

### CODE REVIEW

**Comment by keyman-server** [comment]
This pull request is from an external repo and will not automatically be built. The build must still be passed before it can be merged. Ask one of the team members to make a manual build of this PR.

**Review by ermshiperete — CHANGES_REQUESTED** [review]
LGTM. The only question I have is if this isn't overkill. Another approach would be to remove everything between `<` and `>`, but I'm torn which way is better.

Anyways, this change introduces a new dependency, so we'll have to update the documentation (`docs/linux/keyman-config.md`) and the files that list dependencies (`linux/build/agent/dependencies.config` and `linux/debian/control`).

BeautifulSoup is available as Ubuntu package python3-bs4.

**Comment by TheSchlunz** [comment]
Removing everything between `<` and `>` is actually what I did first but stuff like this still looked bad because some text was just connected afterwards:
image

**Review by ermshiperete — APPROVED** [review]

**Comment by mcdurdin** [comment]
I don't think we can fix the inclusion of HTML tags like this. There is also potential for hyperlinks, images, and other content there.

I think it would be better to consider updates to the .keyboard_info spec to either exclude rich text, or move to Markdown.

**Comment by ermshiperete** [comment]
I just noticed that on Linux we're displaying different text from Windows, so the solution might be to simply change what we display.

**Comment by mcdurdin** [comment]
> I just noticed that on Linux we're displaying different text from Windows, so the solution might be to simply change what we display.

Yes, on Windows, the details from the .keyboard_info / api.keyman.com response are never shown in the keyboard info; only the package description from the .kps is shown. This would be a better solution IMO.

**Review by mcdurdin — CHANGES_REQUESTED** [review]
Change package description to reading data from kmp.json file. (In fact, we should avoid storing the .keyboard_info data, given that is coming from api.keyman.com, which is not available for side-loaded keyboards.)

**Comment by ermshiperete** [comment]
Closing this PR since we're displaying the wrong data in the first place.

**Comment by ermshiperete** [comment]
> Change package description to reading data from kmp.json file. (In fact, we should avoid storing the .keyboard_info data, given that is coming from api.keyman.com, which is not available for side-loaded keyboards.)

@mcdurdin  I'm confused. `kmp.json` doesn't contain anything useful to display for the package description. `khmer_angkor.json` contains a description field, which is what we're already displaying. I can't find the text that Windows displays in the installed keyboard folder. Or are you saying that Windows should do it differently?

**Comment by mcdurdin** [comment]
True, the only thing that Windows shows right now is the "Message" from the .kmx. We plan to merge the .keyboard_info and .kps metadata in 17.0, so let's hold off on these changes for now.

**Pull request closed without merging**
